desc/descriptor_test.go

```diff
@@ -1201,3 +1201,22 @@ func TestToFileDescriptorSet(t *testing.T) {
 		testutil.Eq(t, expectedFile.AsFileDescriptorProto(), f)
 	}
 }
+
+func TestJsonCamelCase(t *testing.T) {
+	testCases := map[string]string{
+		// NB: these example keys were all run through protoc and the
```

Comment:
Does protocolbuffers/protobuf have a similar test and could we use the test cases from there?